p-token/src/processor/approve_checked.rs

```diff
@@ -1,19 +1,21 @@
 use {
-    super::shared,
-    pinocchio::{account_info::AccountInfo, program_error::ProgramError, ProgramResult},
+    super::{shared, U64_BYTES},
+    pinocchio::{account_info::AccountInfo, ProgramResult},
+    spl_token_interface::error::TokenError,
 };
 
 #[inline(always)]
 pub fn process_approve_checked(accounts: &[AccountInfo], instruction_data: &[u8]) -> ProgramResult {
     // expected u64 (8) + u8 (1)
-    let (amount, decimals) = if instruction_data.len() == 9 {
-        let (amount, decimals) = instruction_data.split_at(core::mem::size_of::<u64>());
+    let (amount, decimals) = if instruction_data.len() >= 9 {
+        let (amount, decimals) = instruction_data.split_at(U64_BYTES);
         (
-            u64::from_le_bytes(amount.try_into().unwrap()),
+            // SAFETY: The size of `amount` is `U64_BYTES` bytes.
+            unsafe { u64::from_le_bytes(*(amount.as_ptr() as *const [u8; U64_BYTES])) },
             decimals.first().copied(),
         )
     } else {
-        return Err(ProgramError::InvalidInstructionData);
+        return Err(TokenError::InvalidInstruction.into());
     };
```

Comment:
nit: same here / up to you, but this could also be refactored for easier re-use

Reply:
Good call! Looks nicer with a helper.